### PR TITLE
v7.9.0 (CPD-11491) - Optional removal of console.log statements

### PR DESCRIPTION
**Pull request by xander-marjoram**
Added a value to the `config.js` that can override the removal of console.log() statements that would usually happen automatically for production builds.

Had to revert Jest to 21.2.1 as it was causing problems.

### CODE REVIEW

**Comment by ashleynolan** [comment]
Few conflicts need resolving

**Comment by xander-marjoram** [comment]
@ashleynolan Resolved

File: README.md [hunk 1]
```diff
@@ -506,7 +507,13 @@ Will add a content hash to the JS and CSS filenames, generating a new filename i
 
   Type: `boolean`
 
-  Default is set to false, when set to true this will bundle a versioned JS file e.g `'filename-[version].js'`.
+  Default is set to `false`, when set to true this will bundle a versioned JS file e.g `'filename-[version].js'`.
+
+- #### `stripDebug`
+
+  Type: `boolean`
```
Comment (DamianMullins) [inline README.md]: Add the default type like some of the other options do above.

File: README.md [hunk 2]
```diff
@@ -443,7 +444,7 @@ Will add a content hash to the JS and CSS filenames, generating a new filename i
 
   Type: `boolean`
 
-  Default is set to false, when set to true this will bundle a versioned css file e.g `'filename-[version].css'`.
+  Default is set to `false`, when set to true this will bundle a versioned css file e.g `'filename-[version].css'`.
```
Comment (DamianMullins) [inline README.md]: Looks like this one is also missing the default option..

File: config.js [hunk 3]
```diff
@@ -45,7 +45,8 @@ const ConfigOptions = () => {
             },
             jsDir: 'js',
             lintPaths: [''],
-            usePackageVersion: false
+            usePackageVersion: false,
+            stripDebug: true
```
Comment (DamianMullins) [inline config.js]: Can you add the relevant unit tests to `config.test.js` please?

File: tasks/javascript.js [hunk 4]
```diff
@@ -134,7 +134,7 @@ gulp.task('scripts:bundle', () => {
             ))
 
             // if production build, rip out our console logs
-            .pipe(gulpif(config.isProduction,
+            .pipe(gulpif(config.isProduction && config.js.stripDebug,
```
Comment (kevinrodrigues) [inline tasks/javascript.js]: Might be a silly question but why do we need to have this new option? Shouldn't we strip console logs / debuggers from production code anyway? Or is there a reason we use them in production.

File: README.md (hunk 2 above)
Comment (kevinrodrigues) [reply README.md]: That's my bad, I can fix this in my PR as part of the original ticket unless @xander-marjoram wants to amend it here? :)

File: tasks/javascript.js (hunk 4 above)
Comment (ashleynolan) [reply tasks/javascript.js]: It's a request from the OrderWeb team.  They want to have production type code with debug statements for when they release to QA.

Comment (ashleynolan) [reply tasks/javascript.js]: One thing that would be worth adding is the ability to set this config variable through an environment variable.  So something like `gulp --stripDebug=false` would set this to false (as I imagine that might be how they need to set it to be different on every release server).

File: package.json [hunk 5]
```diff
@@ -71,7 +71,7 @@
     "handlebars-helpers": "^0.10.0",
     "helper-markdown": "^1.0.0",
     "helper-md": "^0.2.2",
-    "jest-cli": "^22.1.4",
+    "jest-cli": "^21.2.1",
```
Comment (DamianMullins) [inline package.json]: For future reference, one of the issues with `v22.1.4` was that an error of "`TypeError: environment.setup is not a function`" was being thrown in some repos, there is a discussion [on the Jest GItHub repo](https://github.com/facebook/jest/issues/5119) where it seems like one of the dependencies hasn't yet been updated,

File: README.md (hunk 2 above)
Comment (xander-marjoram) [reply README.md]: @kevinrodrigues I'll fix it here, if you haven't already 👍

File: config.js (hunk 3 above)
Comment (xander-marjoram) [reply config.js]: Done in latest commit 👍

File: tasks/javascript.js (hunk 4 above)
Comment (xander-marjoram) [reply tasks/javascript.js]: @ashleynolan Should I add that in this PR?

File: test/config.test.js [hunk 6]
```diff
@@ -204,27 +230,42 @@ describe('javascript config', () => {
...
+        expect(config.js.stripDebug).toBeDefined();
     });
 
-    it('`usePackageVersion` should be set to `false` by default', () => {
-        expect(config.js.usePackageVersion).toBe(false);
+    it('`stripDebug` should be set to `true` by default', () => {
+        expect(config.js.stripDebug).toBe(true);
+    });
+
+    it('strip debug can be updated', () => {
```
Comment (kevinrodrigues) [inline test/config.test.js]: `stripDebug`.

Comment (xander-marjoram) [reply test/config.test.js]: @kevinrodrigues Should I change the ones that say 'is production' as well then?

Comment (kevinrodrigues) [reply test/config.test.js]: I think so. There's a mixture of descriptions in this file, usually I add the 'thing' I'm talking about or testing as "`methodName`" `` (with backticks) so it's clear what we're actually referring too in the test itself, I've also found it helps to find that reference quicker when a test fails. :)

Comment (DamianMullins) [reply test/config.test.js]: Let's keep things consistent with these names? All of the existing tests use full, space separated words. I don't mind which one we go with as long as they are consistent :sierra_leone:

File: test/config.test.js [hunk 7]
```diff
@@ -204,27 +230,42 @@ describe('javascript config', () => {
...
+    it('`usePackageVersion` should exist', () => {
+        expect(config.js.usePackageVersion).toBeDefined();
     });
 
-    it('dist file can be updated', () => {
+    it('`usePackageVersion` should be set to `false` by default', () => {
+        expect(config.js.usePackageVersion).toBe(false);
+    });
+
+    it('usePackageVersion can be updated', () => {
```
Comment (DamianMullins) [inline test/config.test.js]: No backticks around `usePackageVersion` here.

File: test/config.test.js (hunk 6 above)
Comment (xander-marjoram) [reply test/config.test.js]: I've chosen to use the variable names as they are found in the config, but with backticks to make it clearer.

**Pull request merged**